components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultDomainObjectFactoryTest.java

```diff
@@ -0,0 +1,256 @@
+
+        when(this.users.getCurrentUser()).thenReturn(null);
+    }
+
+    @Test
+    public void cannotCreatePatientWithNoAccess() throws ComponentLookupException {
+        when(this.patient.getDocument()).thenReturn(null);
+        when(this.access.hasAccess(Right.VIEW, null, null)).thenReturn(false);
+        assertNull(this.mocker.getComponentUnderTest().createPatientSummary(this.patient, this.uriInfo));
+    }
+
+    @Test
+    public void createPatientDocumentExceptionReturnsNull() throws Exception {
+        when(this.patient.getDocument()).thenReturn(null);
+        when(this.access.hasAccess(Right.VIEW, null, null)).thenReturn(true);
+        when(this.documentAccessBridge.getDocument(Matchers.any(DocumentReference.class))).thenThrow(Exception.class);
+        assertNull(this.mocker.getComponentUnderTest().createPatientSummary(this.patient, this.uriInfo));
+    }
+
+    @Test
+    public void createPatientPerformsCorrectly() throws ComponentLookupException, URISyntaxException, Exception
+    {
+        XWikiDocument documentReference = mock(XWikiDocument.class);
+        DocumentReference patientDocument = mock(DocumentReference.class);
+        DocumentReference creatorDocument = mock(DocumentReference.class);
+        UriBuilder uriBuilder = mock(UriBuilder.class);
+        URI uri = new URI("uri");
+        Date creationDate = new Date(1);
+
+        when(this.patient.getDocument()).thenReturn(patientDocument);
+        when(this.access.hasAccess(Right.VIEW, null, patientDocument)).thenReturn(true);
+        when(this.documentAccessBridge.getDocument(patientDocument)).thenReturn(documentReference);
+        when(this.patient.getId()).thenReturn("id");
+        when(this.patient.getExternalId()).thenReturn("externalid");
+        when(this.patient.getReporter()).thenReturn(creatorDocument);
+
+        when(creatorDocument.toString()).thenReturn("creator");
+        when(documentReference.getAuthorReference()).thenReturn(creatorDocument);
+        when(documentReference.getVersion()).thenReturn("version");
+        when(documentReference.getCreationDate()).thenReturn(creationDate);
+        when(documentReference.getDate()).thenReturn(creationDate);
+        when(this.uriInfo.getBaseUriBuilder()).thenReturn(uriBuilder);
+        when(uriBuilder.path(PatientResource.class)).thenReturn(uriBuilder);
+        when(uriBuilder.build("id")).thenReturn(uri);
+
+        PatientSummary patientSummary = mocker.getComponentUnderTest().createPatientSummary(this.patient, this.uriInfo);
+
+        assertEquals("id", patientSummary.getId());
+        assertEquals("externalid", patientSummary.getEid());
+        assertEquals("creator", patientSummary.getCreatedBy());
+        assertEquals("creator", patientSummary.getLastModifiedBy());
+        assertEquals("version", patientSummary.getVersion());
+        assertTrue(patientSummary.getCreatedOn() instanceof DateTime);
+        assertTrue(patientSummary.getLastModifiedOn() instanceof DateTime);
```

Comment:
checking that that the return value is an instanceof DateTime here is unnecessary, since that is the return type of the method and will always be that or null. In other words this could just be `AssertNotNull()` but I think it would be better to check for the exact value that you set up: `creationDate`